src/azul/terraform.py

```diff
@@ -800,11 +800,32 @@ def tf_config(self, app_name):
         # Setting this property using AWS API Gateway extensions to the OpenAPI
         # specification works around this issue.
         #
+        # We also use AWS API extensions to set the API Gateway default
+        # responses since, unlike Terraform, it automatically triggers the
+        # deployment of the REST API when the Gateway responses are added and/or
+        # modified.
+        #
         rest_api = resources['aws_api_gateway_rest_api'][app_name]
         assert 'minimum_compression_size' not in rest_api, rest_api
+        assert 'aws_api_gateway_gateway_response' not in resources, resources
         openapi_spec = json.loads(locals[app_name])
         key = 'x-amazon-apigateway-minimum-compression-size'
         openapi_spec[key] = config.minimum_compression_size
+        openapi_spec['x-amazon-apigateway-gateway-responses'] = {
+            f'DEFAULT_{response_type}': {
+                'responseParameters': {
+                    f'gatewayresponse.header.{k}': v
+                    for k, v in [
+                        ('Content-Security-Policy', "'default-src \'self\''"),
+                        ('X-Content-Type-Options', "'nosniff'"),
+                        ('X-Frame-Options', "'DENY'"),
```

Comment:
This looks odd. A comment should explain why every value is enclosed in single quotes. You may also want to use the patterns that we followed for CSP in the browser TF component and the app.